### PR TITLE
write version string like Urcheon

### PR DESCRIPTION
**Pull request by illwieckz**
build-release: write version string like Urcheon

Keep the possibility to append username (something Urcheon does not) but make
it optional and not enabled by default.

_Note:_ appending username is convenient when modding.

### CODE REVIEW

**Comment by slipher** [comment]
Where does the string go? Can you give an example?

**Comment by illwieckz** [comment]
```sh
…/Unvanquished $ ../unvanquished-release-scripts/build-release -v vm
[…]
Creating archive: …/Unvanquished/build/release/vm_0.52.1-20220213-205227-e971ec0.dpk
```

If `HEAD` was `v0.52.1`, it would produce `vm_0.52.1.dpk`.

**Comment by illwieckz** [comment]
So, I improved the version compute, it now takes care of situations like no git repo, no commit, not version tag.

I'll probably backport the code to Urcheon as this one is simpler and much faster (I had less git knowledge when I did the code in Urcheon).

File: build-release [hunk 1]
```diff
@@ -242,11 +245,53 @@ package () {
...
+	local tag_string='0'
+	local date_string=''
+	local ref_string=''
+	local dirt_string=''
+
+	local git_last_commit="$(git rev-list HEAD 2>/dev/null | head -n 1)"
+
+	if [ -n "${git_last_commit}" ]
+	then
+		local git_last_commit_short="${git_last_commit:0:7}"
```
Comment (slipher) [inline build-release]: Use `git rev-parse --short HEAD` for the short hash

File: build-release [hunk 2]
```diff
@@ -342,17 +387,11 @@ build () {
 
 	if "${write_version_string}"
 	then
-		git_last_commit="$(git rev-list HEAD | head -n 1)"
-		git_last_commit_short="$(git rev-list HEAD | head -n 1 | cut -c1-7)"
-		git_ref="$(git describe --tags --match 'v*' | cut -c2-)"
-		git_last_tag="$(git describe --tags --match 'v*' | cut -f1 -d'-' | cut -c2-)"
-		git_last_date="$(date --date="@$(git log -1 '--pretty=format:%ct')" -u '+%Y%m%d-%H%M%S')"
+		build_version+="$(printVersion)"
```
Comment (slipher) [inline build-release]: Why `+=` ?

Comment (illwieckz) [reply build-release]: Done.

File: build-release (hunk 1 above)
Comment (illwieckz) [reply build-release]: Done.

**Comment by slipher** [comment]
LGTM

**Pull request merged**